# B_anynana_differentialSplicing


## Abstract
Seasonal plasticity is accomplished via tightly regulated developmental cascades that translate environmental cues into trait changes. Little is known about how alternative splicing and other posttranscriptional molecular mechanisms contribute to plasticity or how these mechanisms impact how plasticity evolves. Here, we use transcriptomic and genomic data from the butterfly Bicyclus anynana, a model system for seasonal plasticity, to compare the extent of differential expression and splicing and test how these axes of transcriptional plasticity differ in their potential for evolutionary change. Between seasonal morphs, we find that differential splicing affects a smaller but functionally unique set of genes compared to differential expression. Further, we find strong support for the novel hypothesis that spliced genes are more susceptible than differentially expressed genes to erosion of genetic variation due to selection on seasonal plasticity. Our results suggest that splicing plasticity is especially likely to experience genetic constraints that could affect the potential of wild populations to respond to rapidly changing environments.

## Data availability
RNA-seq data used in to estimate differential splicing and differential expression in this study were accessed from NCBI archives (PRJNA376691, https://www.ncbi.nlm.nih.gov/bioproject/376691). Illumina short-read whole genome data generated in this study and used to calculate nulceotide polymorphism were archived at ___ under accession number ___ (link). Metadata and results of differential expression and spicing analyses have been included in the supplementary data file (.xlsx), as described in the supplementary information file (.pdf). Source data necessary to perform subsequent analyses are provided. 

## Code Availability

Bash and R scripts for all analyses generating lists of differentially spliced and differentially expressed genes are provided here, including: 

* Accessing, cleaning and mapping reads
* Differential splicing (i.e. exon expression) analysis and differential expression analysis in edgeR
* Differential splicing analysis in rMATS
* Gene set enrichment analysis

Data necessary to perform subsequent analyses will be published with the article.
